docs/cardinal/shard/evm-to-cardinal.mdx

```diff
@@ -66,14 +66,14 @@ interface IRouter {
 
 ### sendMessage
 
-The `sendMessage` method enables smart contracts to send messages to the game shard specified by the given namespace.
+The `sendMessage` method enables smart contracts to send messages to the game shard specified by the given namespace. The messageID must be the fully qualified name of the message. For Cardinal game shards, the fully qualified name is of the form "groupName.MessageName". For example, the `create-persona` message's fully qualified name is `persona.create-persona`. Messages without a specified custom group in Cardinal have a default group called "game".
 
 #### Parameters
 
 | Parameter   | Type     | Description                                                                                |
 |-------------|----------|--------------------------------------------------------------------------------------------|
 | `message`   | `bytes`  | ABI encoded message struct.                                                                |
-| `messageID` | `string` | Message identifier. Usually the name passed into Cardinal's `NewMessageTypeWithEVMSupport` |
```

Comment:
We can do this as a separate PR, but I prefer if the function signature is like this instead.

`func sendMessage(message bytes, messageGroup string, messageName string, namespace string)`

Then Cardinal would be the one that turns group + name into the ID.

This is better IMO because 1) user doesn't need to know how to format the ID correctly, 2) less odds of fat finger, 3) if for whatever reason we want to change how we encode the ID internally within Cardinal, they don't have to switch the messageID manually.

The downside of this is that we are imposing an opinion on what the message identifier schema is. In a scenario where let's say we have a Cardinal alternative implementation that doesn't use groups, then it would be awkward. If we wanted to add version as a part of the identifier, then this precompile function signature would have to be updated too.

Curious to hear your thoughts

Reply:
i dont think the user should have to know the ID at all, tbh. nor should they have to worry about fat fingering keys.

the ideal usage of router is with helper smart contracts like this:

```solidity
contract DarkFrontierHelper {
    string sendEnergyID = "game.send-energy"
    struct SendEnergyInput {
        string To;
        string From;
        string PlanetID;
    }


    function SendEnergy(SendEnergyInput input, string namespace) {
        bz = abi.encode(input)
        router.sendMessage(bz, sendEnergyID, namespace)
    }
}
```

i would like to strive to keep router as slim and unopinionated as possible. this does have the downside that smart contracts would have to update if we want to change the schema, but i'm fine with the tradeoff if we get to keep router slim.